libr/include/r_util/r_assert.h

```diff
@@ -0,0 +1,116 @@
+#ifndef R_ASSERT_H
+#define R_ASSERT_H
+
+#include "r_log.h"
+
+R_API void r_assert_log(RLogLevel level, const char *fmt, ...);
+
+#if defined (__GNUC__) && defined (__cplusplus)
+#define R_FUNCTION     ((const char*) (__PRETTY_FUNCTION__))
+#elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#define R_FUNCTION     ((const char*) (__func__))
+#elif defined (__GNUC__) || (defined(_MSC_VER) && (_MSC_VER > 1300))
+#define R_FUNCTION     ((const char*) (__FUNCTION__))
+#else
+#define R_FUNCTION     ((const char*) ("???"))
+#endif
+
+#define r_warn_if_reached()						\
+	do {								\
+		r_assert_log (R_LOG_WARNING, "(%s:%d):%s%s code should not be reached\n", \
+			__FILE__, __LINE__, R_FUNCTION, R_FUNCTION[0] ? ":" : ""); \
+	} while (0)
+
+#define r_warn_if_fail(expr)					\
+	do {								\
+		if (!(expr)) {						\
+			r_assert_log (R_LOG_WARNING, "WARNING (%s:%d):%s%s runtime check failed: (%s)\n", \
+				__FILE__, __LINE__, R_FUNCTION, , R_FUNCTION[0] ? ":" : "", #expr); \
+		}							\
+	} while (0)
+
+/*
+ * R_CHECKS_LEVEL determines the behaviour of the r_return_* set of functions.
+ *
+ * 0: completely disable every function and make them like no-operation
+ * 1: silently enable checks. Check expressions and do return, but do not log anything
+ * 2: enable checks and logging (DEFAULT)
+ * 3: transform them into real assertion
+ */
+#ifndef R_CHECKS_LEVEL
+#define R_CHECKS_LEVEL 2
```

Comment:
who is in charge to set this value? imho a configure flag (./configure --with-checks-level=1), about meson, up2u

so we should probably take the default value from r_userconf.h

Reply:
I tried it simply by adding a `CFLAGS="-DR_CHECKS_LEVEL=3` when doing configure/meson. I think there's no reason at this moment to complicate the build system when there are other ways to set this.

Reply:
adding a configure flag doesnt complicates the build system at all, it just makes this option more visible

Reply:
ok going to do this

Reply:
I can do it with meson, but I have no idea how to do that with acr. Please help.